docs/UsersGuide/best/ground-interface.md

```diff
@@ -0,0 +1,193 @@
+# Ground Interface Architecture and Customization
+
+This guide will discuss the F´ ground interface layers and how to customize them. There are two parts to the ground
+interface: the spacecraft side, and the ground side. This guide will primarily focus on the spacecraft side adaptation
+as the most common pattern is to adapt F´ flight software for some other ground system (Cosmos, OpenMCT, etc). This
+document will walk through common adaptations in hopes that projects will not need to replace the ground interface
+entirely.
+
+In the most basic form, the F´ ground system pattern consists of two sides: uplink and downlink. These two sides each
+have two layers: framing and driver.  Uplink handles data coming from the remote side of the interface, downlink handles
+data going to the remote interface, framing handles serializing and deserializing data to and from byte buffers, and the
+driver layer handles writing data to and from the hardware.
+
+Also of note is the framing protocol, which breaks out the handling of the byte serialization for quick adaptation. Each
+of these stages need to allocate memory and thus users should also consult the [buffer management](./dynamic-memory.md)
+guide.
+
+**Note:** in this guide we will refer to the driver layer but many project will refer to it as the radio or
+communication layer. The function of this layer is to read and write bytes to some hardware and the nature of that
+hardware is irrelevant as long as it can send and receive bytes.
+
+## Ground Interface Architecture
+
+Standard F´ components handle two types of data: com buffers and raw buffers. Com buffers transmit standard F´ items
+(e.g. events, telemetry, and commands) whereas raw buffers (`Fw::Buffer`) transmit any raw bytes like file data. Thus
+the F´ ground interface must handle both types of data. Communications hardware typically only transmits bytes of data
+and knows nothing about the nature of that data. The goal of the ground interfaces is to ensure that the various types
+of F´ data can be translated into a sequence of byte that can be reconstructed on the other side of the interface. This
+architecture is described below.
+
+### Driver
+
+Drivers manager hardware communications. These can be simple hardware interfaces (e.g. TCP or UART) or far more complex
+constructs (e.g. radios, spacecraft buses). From the perspective of F´, the driver has two functions: provide incoming
+data, and handle outgoing data.
+
+**Note:** typically projects use a single driver to handle both input and output, however; two drivers may be used to
+if differing behavior is needed for uplink and downlink.(e.g. UDP downlink for speed and  Tcp uplink reliability).
+
+All drivers implement an input port used to send data out to the hardware. Drivers implement at least one of two methods
```

Comment:
This sentence is similar to the first sentence in the `**Sending Data**` section.
`All drivers implement an input port used to send data out to the hardware.`

Also the verbiage is a little confusion in my opinion. Why an input port is used to send data out?